ynr/apps/popolo/migrations/0034_populate_from_ee_divisions.py

```diff
@@ -0,0 +1,147 @@
+            identifier_label = "osni_oid"
+        if re.match(r"^[A-Z]{3}", identifier_value.split(":")[0]):
+            identifier_label = "dc_slug"
+    else:
+        if re.match(r"[\d]+", post.slug):
+            identifier_label = "mapit_id"
+            identifier_value = post.slug
+        else:
+            identifier_label = "legacy_slug"
+            identifier_value = post.slug
+
+    post.postidentifier_set.update_or_create(
+        identifier=identifier_value,
+        post=post,
+        defaults={"label": identifier_label},
+    )
+
+
+def populate_from_ee_divisions(apps, schema_editor):
+
+    Ballot = apps.get_model("candidates", "Ballot")
+    Post = apps.get_model("popolo", "Post")
+    data_file = open("data/ee_post_identifiers_2020-01.csv")
+
+    for line in csv.DictReader(data_file.readlines()):
+        try:
+            ballot = Ballot.objects.get(ballot_paper_id=line["election_id"])
+        except Ballot.DoesNotExist:
+            continue
+
+        existing_post_qs = Post.objects.filter(
+            start_date=line["start_date"],
+            slug=line["slug"],
+            organization__slug="{}:{}".format(
+                line["organisation_type"], line["org_slug"]
+            ),
+        )
+        if existing_post_qs.exists():
+            post = existing_post_qs.first()
+        else:
+            post = ballot.post
+            post.start_date = line["start_date"]
+        post.end_date = line["end_date"]
+        post.territory_code = line["territory_code"]
+        post.name = line["name"]
+
+        if line["official_identifier"] != post.identifier:
+            update_or_create_post_identifier(post)
+            post.identifier = line["official_identifier"]
+        post.slug = line["slug"]
+
+        post.save()
+        ballot.post = post
+        ballot.save()
+
+
+def populate_from_ee_no_divisions(apps, schema_editor):
+
+    Ballot = apps.get_model("candidates", "Ballot")
+    Post = apps.get_model("popolo", "Post")
+    data_file = open("data/ee_post_identifiers_no_divisions_2020-01.csv")
+
+    for line in csv.DictReader(data_file.readlines()):
+        try:
+            ballot = Ballot.objects.get(ballot_paper_id=line["election_id"])
+        except Ballot.DoesNotExist:
+            continue
+
+        existing_post_qs = Post.objects.filter(
+            start_date=line["start_date"],
+            slug=line["slug"],
+            organization__slug="{}:{}".format(
+                line["organisation_type"], line["slug"]
+            ),
+        )
+        if existing_post_qs.exists():
+            post = existing_post_qs.first()
+        else:
+            post = ballot.post
+            post.start_date = line["start_date"]
+        post.end_date = line["end_date"]
+        post.territory_code = line["territory_code"]
+        post.name = line["name"]
+
+        if line["gss"] != post.identifier:
+            update_or_create_post_identifier(post)
+            post.identifier = line["gss"]
```

Comment:
Setting `post.identifier = line["official_identifier"]` in the case where we do have divisions gives us `gss:X01000001` (if the identifier is a GSS code), but setting `post.identifier = line["gss"]` here in this case gives us `X01000001`. Is that conscious or oversight? It would probably be more consistent for data consumers to also add a `gss:` prefix in this case too.

Reply:
Good spot – pushed a new commit to fix this